src/containers/share-modal/ShareModal.tsx

```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+import { ShareDrawer } from './components/ShareDrawer'
+
+export const ShareModal = () => {
+  return <ShareDrawer />
```

Comment:
is this going to do the switching logic for modal vs. drawer?

Reply:
yeah exactly, just wanted to establish that here! does this seem like the right place for that?

Reply:
A few ways we could do this I guess,
1. Like you've done here (I think fairly reasonable) with the useIsMobile() hook. Maybe worth calling this outer component something less UI-bound than Modal and just like src/containers/Share (ShareView?) - no strong opinion though
2. The provider pattern we've done for page/container level things like this: https://github.com/AudiusProject/audius-client/blob/master/src/containers/track-page/TrackPage.tsx#L19. I think this might be a bit verbose for what you need to do here
3. Some other new HOC component called DrawerOrModal (with a better name maybe) that handles this somewhat common usecase of modal in web, drawer in mobile

Reply:
All great thoughts, I personally like the idea of naming it a more general UI name, cause one is a modal, and one is a drawer (though technically a modal?). I was hoping to try and keep the actual ui components closely tied to the redux state, following https://github.com/AudiusProject/audius-client/blob/master/src/containers/share-sound-to-tiktok-modal/ShareSoundToTikTokModal.tsx, how does that sound?

Reply:
I think that's a good call. No blocker from me here!